test/runtests.jl

```diff
@@ -259,4 +259,28 @@ end
     end
 end
 
+@testset "Product manifold" begin
+    M1 = Manifolds.Sphere(2)
+    M2 = Manifolds.Euclidean(2)
+    M = Manifolds.Product(M1, M2)
+
+    types = [Vector{Float64},
+             SizedVector{5, Float64},
+             MVector{5, Float64},
+             Vector{Float32},
+             SizedVector{5, Float32},
+             MVector{5, Float32}]
+
+    for T in types
+        @testset "Type $T" begin
+            pts = [convert(T, [1.0, 0.0, 0.0, 0.0, 0.0]),
```

Comment:
What I meant in my comment is that this construction of a point (just a 5-element array where the first 3 are on S2, the remaining 2 on Rn is quite complicated to read and hard to understand. I can understand that inernally that might be a good representation and that the views (I think I roughly understood them) are a good help, but would prefer a construction (at least additionally) of the type `([1.0, 0.0, 0.0], [0.0, 0.0])`, the elements on the product manifold are ordered anyways (hence the tuple idea).

Reply:
I favor this approach too. It also enables us to mix and match representations without hassle and doesn't force us to flatten matrices and such. I also think it's clearer in some cases. For example, `SE(n)`'s underlying topological manifold is `SO(n) \times R(n)`. Its elements _can_ be represented as `(n+1) \times (n+1)` matrices which act on elements of `R(n+1)`, but since `SE(n)` acts on elements of `R(n)`, it's cleaner and more efficient to represent its elements as a 2-tuple containing an element of `SO(n)` and one of `R(n)`.

Reply:
This could actually be implemented with a custom `MPoint` called `ProductMPoint`, which stores a list of points and enables dispatch on their types.

Reply:
That would be one way, which might not be efficient (though from modelling side I like it). I like how easy @mateuszbaran can compute logs and so on with his internal single-dimension array representation. If a constructor like I proposed and accessorscan be realized (something like `p[1]`and `p[2]` accessing first and second manifold element of the Product manifold point `p`(and that might require a type to dispatch the array access on, maybe). But I would favour that just for ease of use. I am working a lot with product (and even more with power manifolds, even $ \mathcal M^{128\times 128} $ or something, so I really would like to see easy access to single manifolds values.

Reply:
Yeah, I don't know how efficient this approach is. Still working through @mateuszbaran's array-based approach.

Reply:
I'll definitely add a more user-friendly method of creating points on product manifolds once I'm satisfied with the backend.

Tuples are not that great because they are:

1) Not `AbstractArray`s, which limits compatibility of the representation with some parts of the Julia ecosystem.
2) Immutable.

I'd love to have a mutable heterogeneous `AbstractArray` but it doesn't exist right now. I'm giving it a serious thought though.

I'll definitely add some convenient accessors.

Reply:
I think you can mutate vectors in a tuple of Vectors/Arrays. Certainly, I'm missing context, but this works.
```julia
julia> x = (rand(2), rand(3,3))
([0.178285, 0.857401], [0.525022 0.700751 0.294022; 0.788598 0.253698 0.641965; 0.42019 0.56932 0.875257])

julia> x[1] .= 1
2-element Array{Float64,1}:
 1.0
 1.0

julia> x
([1.0, 1.0], [0.525022 0.700751 0.294022; 0.788598 0.253698 0.641965; 0.42019 0.56932 0.875257])
```

Reply:
Hmm... it didn't work in FunManifolds because I wanted to be able to represent points as `SArray`s but since it's not supported here it might work. I'll try that and see what happens. Things like broadcasting, `representation_size` and  `ambient_space` will be challenging though.